packages/thirdweb/src/transaction/transaction-store.ts

```diff
@@ -0,0 +1,37 @@
+import type { Address } from "../utils/address.js";
+import type { Hex } from "../utils/encoding/hex.js";
+
+type StoredTransaction = {
+  transactionHash: Hex;
+  chainId: number;
+};
+
+const transactionStore = new Map<Address, StoredTransaction[]>();
```

Comment:
Makes sense as a session, but UX wise isn't weird if you refresh and your tx is gone?

We can start with this but wonder if we should still store locally?

Maybe it's not a session to list but a 'local' tx list?

Reply:
What if the API lets you pass a optional storage, and if not passed in stores in memory?

Reply:
needs indexer for this

Reply:
Yeah it's definitely not perfect UX @joaquim-verges but I think the conclusion we landed on was to keep it really light right now and extend with optional storage if needed. Want to avoid splitting sendTransaction implementation between web and native so definitely don't want local storage to be baked into it